flang/lib/Optimizer/Dialect/FIROps.cpp

```diff
@@ -1462,6 +1462,10 @@ void fir::InsertOnRangeOp::build(mlir::OpBuilder &builder,
 
 /// Range bounds must be nonnegative, and the range must not be empty.
 static mlir::LogicalResult verify(fir::InsertOnRangeOp op) {
+  if (fir::hasDynamicSize(op.seq().getType()) ||
+      fir::sequenceWithNonConstantShape(
+          op.seq().getType().cast<fir::SequenceType>()))
```

Comment:
It seems like the second case is an oversight omission from the first test. Can we update `fir::hasDynamicSize()` to return true when the type is an array of unknown rank?

Reply:
Let me try that and see if that works.

Reply:
It's actually already in `fir::hasDynamicSize`. Just removed `fir::sequenceWithNonConstantShape` .